### PR TITLE
Move login process around to provide a better user experience

### PR DESCRIPTION
**Pull request by bundabrg**
# Overview

Move around how the login process works to provide a better user experience.

We do this by keeping the "Finding Server" screen up and preloading the downstream connection simultaneously. Only once things are finalized do we then show the world and spawn in all the cached entities.

Supercedes #656

# To Test
* [X] Offline Login
* [x] Online Login with Floodgate

### CODE REVIEW

**Comment by DoctorMacc** [comment]
Is this ready to go excluding testing?

**Comment by EasyClifton** [comment]
It says, that online login with floodgate has not been tested in the To Test thing.

File: connector/src/main/java/org/geysermc/connector/network/translators/java/JavaJoinGameTranslator.java [hunk 1]
```diff
@@ -74,10 +74,10 @@ public void translate(ServerJoinGamePacket packet, GeyserSession session) {
         session.sendUpstreamPacket(playerGameTypePacket);
         session.setGameMode(packet.getGameMode());
 
-        SetEntityDataPacket entityDataPacket = new SetEntityDataPacket();
-        entityDataPacket.setRuntimeEntityId(entity.getGeyserId());
-        entityDataPacket.getMetadata().putAll(entity.getMetadata());
-        session.sendUpstreamPacket(entityDataPacket);
+//        SetEntityDataPacket entityDataPacket = new SetEntityDataPacket();
```
Comment (DoctorMacc) [inline connector/src/main/java/org/geysermc/connector/network/translators/java/JavaJoinGameTranslator.java:77]: Can you remove this?

**Comment by DoctorMacc** [comment]
- Online login with credentials not saved works
- Online login with credentials saved works
- Offline login works
- Floodgate works

**Comment by lukeeey** [comment]
!start-test-server

**Comment by LucasMucGH** [comment]
!start-test-server

**Comment by Camotoy** [comment]
Seeing as this PR is a year old and the following description of the PR is now implemented in Geyser since 1.18:

`We do this by keeping the "Finding Server" screen up and preloading the downstream connection simultaneously.`

I'm going to go ahead and close this. Thanks anyway, of course!

**Pull request closed without merging**